File: NEsper/events/ReflectionPropFieldGetter.cs
using System;
using System.Reflection;

namespace net.esper.events
{
	/// <summary>
    /// Property getter for fields using Java's vanilla reflection.
    /// </summary>
	
    public sealed class ReflectionPropFieldGetter : EventPropertyGetter
	{
		private readonly FieldInfo field;
		
		/// <summary> Constructor.</summary>
		/// <param name="field">is the regular reflection field to use to obtain values for a property
		/// </param>
		
        public ReflectionPropFieldGetter(FieldInfo field)
		{
			this.field = field;
		}

        /// <summary>
        /// Gets the <see cref="System.Object"/> with the specified bean as the key.
        /// </summary>
        /// <value></value>

        public Object GetValue(EventBean obj)
        {
            Object underlying = obj.Underlying;

            try
            {
                return field.GetValue(underlying);
            }
            catch (ArgumentException)
            {
                throw new PropertyAccessException("Mismatched getter instance to event bean type");
            }
            catch (UnauthorizedAccessException e)
            {
                throw new PropertyAccessException(e);
            }
            catch (Exception e)
            {
                throw new PropertyAccessException(e);
            }
        }
		
		public override String ToString()
		{
			return "ReflectionPropFieldGetter " + "field=" + field.ToString();
		}
	}
}

File: trunk/esper/src/main/java/net/esper/event/ReflectionPropFieldGetter.java
package net.esper.event;

import java.lang.reflect.Field;

/**
 * Property getter for fields using Java's vanilla reflection.
 */
public final class ReflectionPropFieldGetter implements EventPropertyGetter
{
    private final Field field;

    /**
     * Constructor.
     * @param field is the regular reflection field to use to obtain values for a property
     */
    public ReflectionPropFieldGetter(Field field)
    {
        this.field = field;
    }

    public final Object get(EventBean obj) throws PropertyAccessException
    {
        Object underlying = obj.getUnderlying();

        try
        {
            return field.get(underlying);
        }
        catch (IllegalArgumentException e)
        {
            throw new PropertyAccessException("Mismatched getter instance to event bean type");
        }
        catch (IllegalAccessException e)
        {
            throw new PropertyAccessException(e);
        }
    }

    public String toString()
    {
        return "ReflectionPropFieldGetter " +
                "field=" + field.toGenericString();
    }

    public boolean isExistsProperty(EventBean eventBean)
    {
        return true; // Property exists as the property is not dynamic (unchecked)
    }    
}
